schema.json

```diff
@@ -101,58 +103,51 @@
             "type": "string",
             "format": "regex"
         },
-        "additionalItems": { "$ref": "#" },
+        "additionalItems": { "$recursiveRef": "#" },
         "items": {
             "anyOf": [
-                { "$ref": "#" },
+                { "$recursiveRef": "#" },
                 { "$ref": "#/$defs/schemaArray" }
-            ]
+            ],
+            "default": true
         },
         "maxItems": { "$ref": "#/$defs/nonNegativeInteger" },
         "minItems": { "$ref": "#/$defs/nonNegativeIntegerDefault0" },
         "uniqueItems": {
             "type": "boolean",
             "default": false
         },
-        "contains": { "$ref": "#" },
+        "contains": { "$recursiveRef": "#" },
         "maxProperties": { "$ref": "#/$defs/nonNegativeInteger" },
         "minProperties": { "$ref": "#/$defs/nonNegativeIntegerDefault0" },
         "required": { "$ref": "#/$defs/stringArray" },
-        "additionalProperties": { "$ref": "#" },
+        "additionalProperties": { "$recursiveRef": "#" },
         "properties": {
             "type": "object",
-            "additionalProperties": { "$ref": "#" },
+            "additionalProperties": { "$recursiveRef": "#" },
             "default": {}
         },
         "patternProperties": {
             "type": "object",
-            "additionalProperties": { "$ref": "#" },
+            "additionalProperties": { "$recursiveRef": "#" },
             "propertyNames": { "format": "regex" },
             "default": {}
         },
-        "dependentSchemas": {
+        "dependencies": {
             "type": "object",
             "additionalProperties": {
-                "$ref": "#"
+                "$recursiveRef": "#"
             }
         },
-        "dependentRequired": {
+        "requiredDependencies": {
             "type": "object",
             "additionalProperties": {
                 "$ref": "#/$defs/stringArray"
             }
         },
-        "dependencies": {
-            "$comment": "\"dependencies\" is no longer a keyword, but schema authors should avoid redefining it to facilitate a smooth transition to \"dependentSchemas\" and \"dependentRequired\"",
-            "type": "object",
-            "additionalProperties": {
-                "anyOf": [
-                    { "$ref": "#" },
-                    { "$ref": "#/$defs/stringArray" }
-                ]
-            }
+        "propertyNames": {
```

Comment:
This change doesn't fit the wording of the `propertyNames` keyword.

>The value of "propertyNames" MUST be a valid JSON Schema.
>
>If the instance is an object, this keyword validates if every property name in
>the instance validates against the provided schema.
>Note the property name that the schema is testing will always be a string.
>
>Omitting this keyword has the same behavior as an empty schema.

Reply:
I wonder if there's a way that we can write a schema that validates schemas that validate strings.  That would be perfect here.

```
{
  "allOf": [
    { "$recursiveRef": "#" },
    {
      "type": "object",
      "properties": {
        "type": {"const":"string"}
      },
      "required": [ "type" ]
    }
  ]
}
```

Concise? No. Explicit? Yes.  Necessary? ¯\_(ツ)_/¯

Reply:
This change was also unintentional.  Generally, we make the meta-schema more permissive rather than trying to lock it down.  We've gone over similar changes before and always decided not to do them.  It's totally valid to write `propertyNames` without type (JSON is not capable of serializing anything else, so its utility is debatable), or even with the wrong type (although I have no idea why, or what useful thing it could do).  There's a similar restriction with `hrefSchema` that it probably should have `type: object` but we don't enforce that either.